feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts

```diff
@@ -63,39 +66,45 @@ export class B2BUserEffects {
   createB2BUser$: Observable<
     | B2BUserActions.CreateB2BUserSuccess
     | B2BUserActions.CreateB2BUserFail
+    | OrgUnitActions.AssignApprover
     | OrganizationActions.OrganizationClearData
   > = this.actions$.pipe(
     ofType(B2BUserActions.CREATE_B2B_USER),
     map((action: B2BUserActions.CreateB2BUser) => action.payload),
     filter((payload) => isValidUser(payload.userId)),
-    switchMap((payload) =>
-      this.b2bUserConnector.create(payload.userId, payload.orgCustomer).pipe(
+    switchMap(({ userId, orgCustomer }) =>
+      this.b2bUserConnector.create(userId, orgCustomer).pipe(
         switchMap((data) => {
+          const isAssignedToApprovers = orgCustomer.isAssignedToApprovers;
           // TODO Workaround for not known customerId while user creation (redireciton)
           return this.routingService.getRouterState().pipe(
             take(1),
-            tap((route) => {
-              if (
-                (route as any)?.state?.context?.id !== '/organization/units'
-              ) {
-                this.routingService.go({
-                  cxRoute: 'userDetails',
-                  params: data,
-                });
-              }
-            }),
+            tap((route) => this.redirect(route, data)),
             switchMap(() => {
-              return [
+              const successActions = [
                 new B2BUserActions.CreateB2BUserSuccess(data),
                 new OrganizationActions.OrganizationClearData(),
-              ];
+              ] as any[];
+              if (isAssignedToApprovers) {
+                successActions.splice(
+                  1,
+                  0,
```

Comment:
does action order make sense, or it can be just pushed at the end?

Reply:
switchMap does not preserve the order, concatMap does

Reply:
I think this is fine as is